config/initializers/fips.rb

```diff
@@ -0,0 +1,46 @@
+require "openssl"
+require "digest"
+require "digest/md5"
+require "digest/sha1"
+require 'sprockets'
+require 'openid_connect'
+
+# Supress warning messages.
+original_verbose, $VERBOSE = $VERBOSE, nil
+# Remove pre-existing constants if they do exist to reduce the
+# amount of log spam and warnings.
+Digest.send(:remove_const, "MD5") if Digest.const_defined?("MD5")
+Digest.const_set("MD5", OpenSSL::Digest::MD5)
+Digest.send(:remove_const, "SHA1") if Digest.const_defined?("SHA1")
+Digest.const_set("SHA1", OpenSSL::Digest::SHA1)
+Digest.send(:remove_const, "SHA256") if Digest.const_defined?("SHA256")
+Digest.const_set("SHA256", OpenSSL::Digest::SHA256)
+
+# override the default Digest with OpenSSL::Digest
+Digest::SHA256 = OpenSSL::Digest::SHA256
```

Comment:
What is the significance of this line versus the lines above?
```
Digest.send(:remove_const, "SHA256") if Digest.const_defined?("SHA256")
Digest.const_set("SHA256", OpenSSL::Digest::SHA256)
```

Reply:
Chef mentions that
# Remove pre-existing constants if they do exist to reduce the
# amount of log spam and warnings.

https://github.com/chef/chef/blob/4d9857dd8ce9f8f199722c7abad2bdd60a6277d3/chef-config/lib/chef-config/config.rb#L1248
chef-config/lib/chef-config/config.rb:1248

Reply:
Do you have an example of the log spam and warnings this removes in Conjur?

Reply:
It was added only because of the details in the link above.
Though, I have tried it without these lines and there are no extra warnings.
Do you think it is better to remove these lines?

Reply:
It would be worth having others from @cyberark/conjur-core-team weigh in, but if we don't observe the extra warnings these are intended to fix, then I don't think we should include these.

Reply:
@shaharglazner hu,
What do you say regarding the Chef documentation.
I agree with Micah and think it should be removed.

Reply:
Remove the Chef recommendation code after approving with Shahar